pkg/controller/namespace/deletion/status_condition_utils_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deletion
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestProcessContentTotals(t *testing.T) {
+	tests := []struct {
+		name string
+
+		contentTotals allGVRDeletionMetadata
+		expecteds     []v1.NamespaceCondition
+	}{
+		{
+			name: "nothing",
+
+			contentTotals: allGVRDeletionMetadata{
+				gvrToNumRemaining:        map[schema.GroupVersionResource]int{},
+				finalizersToNumRemaining: map[string]int{},
+			},
+			expecteds: []v1.NamespaceCondition{},
+		},
+		{
+			name: "just remaining",
+
+			contentTotals: allGVRDeletionMetadata{
+				gvrToNumRemaining: map[schema.GroupVersionResource]int{
+					{Group: "apps.k8s.io", Resource: "daemonsets"}:  5,
+					{Group: "apps.k8s.io", Resource: "deployments"}: 5,
+				},
+				finalizersToNumRemaining: map[string]int{},
+			},
+			expecteds: []v1.NamespaceCondition{
+				{Type: v1.NamespaceContentRemaining, Status: v1.ConditionTrue, Reason: "SomeResourcesRemain", Message: `Some resources are remaining: daemonsets.apps.k8s.io has 5 resource instances, deployments.apps.k8s.io has 5 resource instances`},
+			},
+		},
+		{
+			name: "just finalizers", // this shouldn't happen
+
+			contentTotals: allGVRDeletionMetadata{
+				gvrToNumRemaining: map[schema.GroupVersionResource]int{},
+				finalizersToNumRemaining: map[string]int{
+					"service-catalog": 6,
+					"kubedb":          5,
+				},
+			},
+			expecteds: []v1.NamespaceCondition{
+				{Type: v1.NamespaceFinalizersRemaining, Status: v1.ConditionTrue, Reason: "SomeFinalizerRemain", Message: `Some finalizers are remaining: kubedb in 5 resource instances, service-catalog in 6 resource instances`},
+			},
+		},
+		{
+			name: "both",
+
+			contentTotals: allGVRDeletionMetadata{
+				gvrToNumRemaining: map[schema.GroupVersionResource]int{
+					{Group: "apps.k8s.io", Resource: "daemonsets"}:  5,
+					{Group: "apps.k8s.io", Resource: "deployments"}: 5,
+				},
+				finalizersToNumRemaining: map[string]int{
+					"service-catalog": 6,
+					"kubedb":          5,
+				},
+			},
+			expecteds: []v1.NamespaceCondition{
+				{Type: v1.NamespaceContentRemaining, Status: v1.ConditionTrue, Reason: "SomeResourcesRemain", Message: `Some resources are remaining: daemonsets.apps.k8s.io has 5 resource instances, deployments.apps.k8s.io has 5 resource instances`},
```

Comment:
I'm a little worried about how big this message can get :/

Reply:
> I'm a little worried about how big this message can get :/

Given that we're only reporting numbers, I think we'll be below a limit.